memlimit/memlimit.go

```diff
@@ -29,6 +30,48 @@ var (
 	logger = log.New(io.Discard, "", log.LstdFlags)
 )
 
+type config struct {
+	printf   func(string, ...interface{})
+	ratio    float64
+	provider Provider
+	env      bool
+}
+
+// An Option alters the behavior of SetGoMemLimitWithEnv.
+type Option func(cfg *config)
+
+// WithLogger uses the supplied printf implementation for log output.
+// By default log.Printf.
+func WithLogger(printf func(string, ...interface{})) Option {
+	return Option(func(cfg *config) {
+		cfg.printf = printf
+	})
+}
+
+// WithRatio configure memory limit.
+// By default `defaultAUTOMEMLIMIT`.
+func WithRatio(ratio float64) Option {
+	return Option(func(cfg *config) {
+		cfg.ratio = ratio
+	})
+}
+
+// WithEnv configure memory limit from environment variable.
+// By default `false`.
+func WithEnv() Option {
+	return Option(func(cfg *config) {
+		cfg.env = true
+	})
+}
```

Comment:
It seems like `WithRatio` and `WithEnv` work exclusively for each other.
What about not adding `WithEnv` and using the environment variable if `WithRatio` is not used?
(i.e. using the environment variable by default, and if the ratio is provided, using it instead.)